packages/rollup-full-node/src/app/web3-rpc-handler.ts

```diff
@@ -755,21 +761,26 @@ export class DefaultWeb3Handler
     )
 
     const res = await this.context.provider.sendTransaction(signedTx)
-    try {
-      const receipt = await this.context.provider.waitForTransaction(res.hash)
-      log.debug(
-        `Got receipt mapping ${ovmTxHash} to ${internalTxHash}: ${JSON.stringify(
-          receipt
-        )}`
-      )
-    } catch (e) {
-      logError(
-        log,
-        `Error mapping ovmTxHash: ${ovmTxHash} to internal tx hash: ${internalTxHash}`,
-        e
-      )
-      throw e
-    }
+    this.ovmHashToOvmTransactionCache[ovmTxHash] = signedOvmTransaction
+
+    this.context.provider
+      .waitForTransaction(res.hash)
+      .then((receipt) => {
+        log.debug(
+          `Got receipt mapping ${ovmTxHash} to ${internalTxHash}: ${JSON.stringify(
+            receipt
+          )}`
+        )
+        delete this.ovmHashToOvmTransactionCache[ovmTxHash]
```

Comment:
Ah nice I was wondering if we'd run into any problems with the cache being too large but this looks good!